common/autoware_overlay_rviz_plugin/autoware_minimap_overlay_rviz_plugin/src/goal_pose.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "include/goal_pose.hpp"
+
+#include <QDebug>
+
+#include <qpoint.h>
+
+GoalPose::GoalPose()
+{
+  std::string package_path =
+    ament_index_cpp::get_package_share_directory("autoware_minimap_overlay_rviz_plugin");
+  std::string goal_image_path = package_path + "/icons/goal.png";
+  goal_image_ = QImage(goal_image_path.c_str());
+  goal_image_ = goal_image_.scaled(20, 20, Qt::KeepAspectRatio);
+}
+
+void GoalPose::setGoalPosition(double local_x, double local_y, double origin_lat, double origin_lon)
+{
+  std::tie(goal_lat_, goal_lon_) = localToGeographic(local_x, local_y, origin_lat, origin_lon);
+}
+
+void GoalPose::setProjectionInfo(const std::string & projector_type, const std::string & mgrs_grid)
+{
+  projector_type_ = projector_type;
+  mgrs_grid_ = mgrs_grid;
+}
+
+std::pair<double, double> GoalPose::localToGeographic(
+  double local_x, double local_y, double origin_lat, double origin_lon)
+{
+  if (projector_type_ == "MGRS") {
+    return localToGeographicMGRS(local_x, local_y, origin_lat, origin_lon, mgrs_grid_);
+  } else {
+    return localToGeographicUTM(local_x, local_y, origin_lat, origin_lon);
+  }
```

Comment:
It's probably better to use the [variables](https://github.com/tier4/tier4_autoware_msgs/blob/a6d8211121615793fb8fdbe3a04a4acea33a09bb/tier4_map_msgs/msg/MapProjectorInfo.msg#L5) defined in the message instead of raw text.

```suggestion
  if (projector_type_ == tier4_map_msgs::msg::MapProjectorInfo::MGRS) {
    return localToGeographicMGRS(local_x, local_y, origin_lat, origin_lon, mgrs_grid_);
  } else if (projector_type_ == tier4_map_msgs::msg::MapProjectorInfo::LOCAL_CARTESIAN_UTM) {
    return localToGeographicUTM(local_x, local_y, origin_lat, origin_lon);
  }
```